File: LeetCode/560. 和为K的子数组.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function (nums, k) {
  const m = new Map();
  m.set(0, 1);
  let c = 0,
    pre = 0;
  for (let x of nums) {
    pre += x;
    if (m.has(pre - k)) c += m.get(pre - k);
    if (m.has(pre)) m.set(pre, m.get(pre) + 1);
    else m.set(pre, 1);
  }
  return c;
};

File: LeetCode/560. 和为K的子数组.rs
impl Solution {
  pub fn subarray_sum(nums: Vec<i32>, k: i32) -> i32 {
      let mut m=std::collections::HashMap::new();
      m.insert(0,1);
      let mut c=0;
      let mut pre=0;
      for x in nums{
          pre+=x;
          c+=m.get(&(pre-k)).unwrap_or(&0);
          *m.entry(pre).or_insert(0)+=1;
      }
      return c;
  }
}